src/popup.js

```diff
@@ -75,27 +25,51 @@ const createPopup = ( container, win = window ) => {
 		popup.currentTargetElement = null
 	}
 
-	const show = ( content, nextTo, pointerPosition ) => {
+	const show = ( content, nextTo, { x: mouseX, y: mouseY } ) => {
 		popup.innerHTML = content
-
-		const position = computePopupPosition(
-			getTargetRect( nextTo, pointerPosition ),
-			popup.offsetWidth,
-			popup.offsetHeight,
-			win.innerWidth,
-			win.innerHeight
-		)
-
-		popup.style.left = withPx( position.left )
-		popup.style.top = withPx( position.top )
-		popup.style.bottom = withPx( position.bottom )
-
-		popup.currentTargetElement = nextTo
-		popup.style.visibility = 'visible'
-
-		if ( popupEvents.onShow ) {
-			popupEvents.onShow( popup )
-		}
+		const arrowEl = popup.querySelector( '.wikipediapreview-arrow' )
+
+		computePosition( nextTo, popup, {
+			middleware: [
+				inline( { x: mouseX, y: mouseY } ),
+				shift(),
+				autoPlacement(),
+				offset( 10 ),
+				arrow( { element: arrowEl } )
+			]
+		} ).then( ( { x, y, middlewareData, placement } ) => {
```

Comment:
I this that this is async with a promise.
1. We've not used promises in this codebase at all for compat reasons. This is probably fine now but we'll have to re-evaluate our supported browsers list and redo some compat testing. This is just a note.
2. Is there a sync interface we could use instead to make it easier to integrate with the reef branch?

Reply:
good point, the source code uses a lot async await code